### PR TITLE
Add support for KHR_texture_transform

### LINKED ISSUE #6335: glTF not being rendered correctly. UV texture issue?
## Bevy version

bevy = "0.8.1"

## What you did

Created an image of a plane in blender with 100x100 m size. Gave a texture for each 1x1meter.
image

image

Bevy has no clue UV isn't the whole plane.

```rs
    let grid_plane = asset_server.load("models\\planewgrid.gltf#Scene0");

    // plane
    commands
        .spawn_bundle(SceneBundle {
            scene: grid_plane,
            ..Default::default()
        })
        .insert(Collider::cuboid(100.0, 0.1, 100.0));
```

## What went wrong

This is what breaks in bevy:

image

## Additional information

Here's the files:
http://gofile.me/6dlZG/cQxW3Sc6H

### PR DESCRIPTION
**Pull request by yrns**
# Objective

Support the KHR_texture_transform extension for the glTF loader.

- Fixes #6335

## Solution

As is, this only supports a single transform. Looking at Godot's source, they support one transform with an optional second one for detail, AO, and emission. glTF specifies one per texture. The public domain materials I looked at seem to share the same transform. So maybe having just one is acceptable for now. I tried to include a warning if multiple different transforms exist for the same material.

Note the gltf crate doesn't expose the texture transform for the normal and occlusion textures, which it should, so I just ignored those for now.

Via `cargo run --release --example scene_viewer ~/src/clone/glTF-Sample-Models/2.0/TextureTransformTest/glTF/TextureTransformTest.gltf`:

texture_transform

## Changelog

Support for the [KHR_texture_transform](https://github.com/KhronosGroup/glTF/tree/main/extensions/2.0/Khronos/KHR_texture_transform) extension added.

### CODE REVIEW

**Comment by mockersf** [comment]
message above is due to a network error on crates.io

**Comment by mockersf** [comment]
Your screenshot doesn't look exactly like the expected (https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/TextureTransformTest), it seems the background for out of texture quads are not the same. Do you know why?

How would it work with https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/TextureTransformMultiTest?

**Comment by yrns** [comment]
> Your screenshot doesn't look exactly like the expected (https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/TextureTransformTest), it seems the background for out of texture quads are not the same. Do you know why?

The default camera and lighting in the scene viewer make it hard to see. The background in the arrow texture is there, just very faint, if that's what you mean.

> How would it work with https://github.com/KhronosGroup/glTF-Sample-Models/tree/master/2.0/TextureTransformMultiTest?

multi_transform

This looks far from complete because the loader doesn't support texture coord sets. I see there's some TODOs in the source code. The renderer doesn't do clearcoat, and the gltf crate doesn't expose a texture transform for occlusion.

File: crates/bevy_gltf/src/loader.rs [hunk 1]
```diff
@@ -627,19 +627,35 @@ async fn load_texture<'a>(
...
+            // TODO: handle info.tex_coord() (the *set* index for the right texcoords)
+            let label = texture_label(&info.texture());
+            let path = AssetPath::new_ref(load_context.path(), Some(&label));
+            (
+                load_context.get_handle(path),
+                info.texture_transform().map(texture_transform_mat3),
+            )
+        })
+        .unzip();
+    let uv_transform = uv_transform.flatten().unwrap_or_default();
```
Comment (nicopap) [inline crates/bevy_gltf/src/loader.rs:658]: Nit: For code quality, I would prefer if the `base_color_texture` and `uv_transform` definitions were split. the `map(…).unzip()` is fairly hard to parse.

**Review by nicopap — APPROVED** [review]
Looks good to me.

Need to be aware of costs though:
1. This adds 36 bytes to the `StandardMaterial` uniform.
2. This is an additional 3x3 matrix multiplication per standard material pixel.

With #9254, the cost of (1) is fairly low now though.

**Comment by mockersf** [comment]
A `Affine2` seems better (smaller and faster) for the transform than a `Mat3`?

**Comment by hchockarprasad** [comment]
Hi… Any update on this??

**Comment by janhohenheim** [comment]
Closes #11869

**Comment by janhohenheim** [comment]
I'm currently updating this PR to `main` and addressing the comments on my fork :)

**Comment by janhohenheim** [comment]
Done, just got some questions on the new PR

**Comment by alice-i-cecile** [comment]
Closing as adopted :)
